gpu_ocean/SWESimulators/GPUDrifterCollection.py

```diff
@@ -43,48 +43,56 @@ def __init__(self, cl_ctx, numDrifters, \
                                 domain_size_x=domain_size_x, 
                                 domain_size_y=domain_size_y)
         
-        # Define OpenCL environment:
-        self.cl_ctx = cl_ctx
+        # Define CUDA environment:
+        self.gpu_ctx = gpu_ctx
         self.block_width = block_width
         self.block_height = 1
         
         # TODO: Where should the cl_queue come from?
         # For sure, the drifter and the ocean simulator should use 
         # the same queue...
-        self.cl_queue = cl_queue
-        if self.cl_queue is None:
-            self.cl_queue = pyopencl.CommandQueue(self.cl_ctx)
-        
+        self.gpu_stream = gpu_stream
+        if self.gpu_stream is None:
+            self.gpu_stream = cuda.Stream()
```

Comment:
This should instead be the default argument, stream=cuda.Stream(). Right now, if I want to pass no stream (none), then I get a stream nonetheless

Reply:
Good idea. That would be cleaner. Sadly, it gives me an exception:
```File "../SWESimulators/GPUDrifterCollection.py", line 37, in GPUDrifterCollection
    gpu_stream=cuda.Stream(), \
pycuda._driver.LogicError: explicit_context_dependent failed: invalid device context - no currently active context?
```
I don't think the current implementation is a problem, though. Without a stream, you obviously can't do anything with the class until you attach the drifter to a Simulator. The gpu_stream in the Simulator object will then replace the stream created in line 54. The question is: How bad is it to create streams that are overwritten? The overwritten streams should go out of scope and be cleaned up by the garbage collector...